CIPs/0003.md

```diff
@@ -0,0 +1,134 @@
+# CIP [0003]: Identity Claims and Metadata
+
+- Date: 2019-09-19
+- Author: @timmoreton
+- Status: WIP
+
+## Overview
+
+This CIP describes a general mechanism that allows principals on the Celo platform to claim real-world identities and maintain rich, unverified metadata. Metadata may be public, or optionally, encrypted and accessible only to the Celo Foundation. Third parties can verify these identity claims, and use the associated unverified metadata as appropriate. 
+
+There are a number of principals that provide resources to or deploy services on the Celo platform that may wish to assert identities or provide metadata. These include, but are not limited to:
+* Validators
+* Validator groups
+* Smart contract owners
+* DApp developers
+* Creators of Governance proposals
+* Oracles 
+
+Use cases include:
+* Tools want to present public metadata supplied by a validator or validator group as part of a list of candidate groups, or a list of current elected validators.
+* Governance Explorer UIs may want to present public metadata about the creators of governance proposals
+* The Celo Foundation receives notice of a security vulnerability and wants to contact elected validators to facilitate them to make a decision on applying a patch.
+* A DApp makes a request to the Celo Wallet for account information or to sign a transaction. The Celo Wallet should provide information about the DApp to allow the user to make a decision whether to accept the transaction or not.   
+
+## Goals
+
+- Integrate with the existing Attestations structure, rather than maintain separate identifer fields in other contracts, so that the same identity claims and metadata can be linked from any of a principals' activities  
+- Allow principals to make an extensible range of identity claims, and support clients verifying the veracity of these 
+- Allow principals to supply unverified extensible, structured metadata off-chain
+- Allow tools to choose what metadata they require and how they present it
+- Support both public (unencrypted) metadata and metadata accessible only to the Celo Foundation
+
+## Proposed Solution
+
+### Claims
+
+The Attestations contract is extended as follows:
+*  Add a `struct Claim` that comprises: 
+   * `service` - Required. Initially supported values for `service` include `primary`, `github`, and `twitter`.
+   * `identifier` - Optional. The identifier being claimed on the service being claimed.  
```

Comment:
I would be in favor of encoding the `service` into the identifier itself. So for phone numbers it would be `tel://+49301112223333`, twitter it would be `twitter://nambrot` and for domains it would be `dns://nambrot.com`, because for attestations, issuers could just attest to the identifier without having to separately specify the service.

Reply:
This can be nice.